cdap-ui/cypress/integration/pipeline.studio.spec.ts

```diff
@@ -0,0 +1,147 @@
+
+  it('Should be able to build a complex pipeline', () => {
+    cy.visit('/pipelines/ns/default/studio');
+    // Two BigQuery sources
+    const sourceNode1: INodeInfo = { nodeName: 'BigQueryTable', nodeType: 'batchsource' };
+    const sourceNodeId1: INodeIdentifier = { ...sourceNode1, nodeId: '0' };
+    const sourceNode2: INodeInfo = { nodeName: 'BigQueryTable', nodeType: 'batchsource' };
+    const sourceNodeId2: INodeIdentifier = { ...sourceNode2, nodeId: '1' };
+
+    // Two javascript transforms 
+    const transformNode1: INodeInfo = { nodeName: 'JavaScript', nodeType: 'transform' };
+    const transformNodeId1: INodeIdentifier = { ...transformNode1, nodeId: '2' };
+    const transformNode2: INodeInfo = { nodeName: 'JavaScript', nodeType: 'transform' };
+    const transformNodeId2: INodeIdentifier = { ...transformNode2, nodeId: '3' };
+
+
+    // One joiner
+    const joinerNode: INodeInfo = { nodeName: 'Joiner', nodeType: 'batchjoiner' };
+    const joinerNodeId: INodeIdentifier = { ...joinerNode, nodeId: '4' };
+
+    // One condition node
+    const conditionNode: INodeInfo = { nodeName: 'Conditional', nodeType: 'condition' }
+    const conditionNodeId: INodeIdentifier = { ...conditionNode, nodeId: '5' };
+
+    // Two BigQuery sinks
+    const sinkNode1: INodeInfo = { nodeName: 'BigQueryMultiTable', nodeType: 'batchsink' };
+    const sinkNodeId1: INodeIdentifier = { ...sinkNode1, nodeId: '6' };
+    const sinkNode2: INodeInfo = { nodeName: 'BigQueryMultiTable', nodeType: 'batchsink' };
+    const sinkNodeId2: INodeIdentifier = { ...sinkNode2, nodeId: '7' };
+
+    cy.add_node_to_canvas(sourceNode1);
+    cy.add_node_to_canvas(sourceNode2);
+
+    cy.open_transform_panel();
+    cy.add_node_to_canvas(transformNode1);
+    cy.add_node_to_canvas(transformNode2);
+
+    cy.open_analytics_panel();
+    cy.add_node_to_canvas(joinerNode);
+
+    cy.open_condition_and_actions_panel();
+    cy.add_node_to_canvas(conditionNode);
+
+    cy.open_sink_panel();
+    cy.add_node_to_canvas(sinkNode1);
+    cy.add_node_to_canvas(sinkNode2);
+
+    cy.get('[data-cy="pipeline-clean-up-graph-control"]').click();
+    cy.get('[data-cy="pipeline-fit-to-screen-control"]').click();
+
+    cy.connect_two_nodes(sourceNodeId1, transformNodeId1, getGenericEndpoint);
+    cy.connect_two_nodes(sourceNodeId2, transformNodeId2, getGenericEndpoint);
+
+    cy.connect_two_nodes(transformNodeId1, joinerNodeId, getGenericEndpoint);
+    cy.connect_two_nodes(transformNodeId2, joinerNodeId, getGenericEndpoint);
+
+    cy.connect_two_nodes(joinerNodeId, conditionNodeId, getGenericEndpoint);
+
+    cy.connect_two_nodes(conditionNodeId, sinkNodeId1, getConditionNodeEndpoint, { condition: true });
+    cy.connect_two_nodes(conditionNodeId, sinkNodeId2, getConditionNodeEndpoint, { condition: false });
+
+    cy.get('[data-cy="pipeline-clean-up-graph-control"]').click();
+    cy.get('[data-cy="pipeline-fit-to-screen-control"]').click();
+
+    cy.compareSnapshot('test');
```

Comment:
is this intended?

Reply:
Changed the name of the file to correspond to something meaningful.